TLM/TLM/Patch/PatchCommons.cs

```diff
@@ -108,76 +107,31 @@ public static class PatchCommons {
             return codes;
         }
 
-        private static List<CodeInstruction> GetUpdatePositionInstructions(Label label) {
-            PropertyInfo getFactory = typeof(Constants).GetProperty(nameof(Constants.ManagerFactory));
-            PropertyInfo getExtVehMan = typeof(IManagerFactory).GetProperty(nameof(IManagerFactory.ExtVehicleManager));
-            MethodInfo updateVehPos = typeof(IExtVehicleManager).GetMethod( nameof(IExtVehicleManager.UpdateVehiclePosition));
-            FieldInfo advAIField = typeof(Options).GetField(nameof(Options.advancedAI));
-            FieldInfo vehFlags = typeof(Vehicle).GetField(nameof(Vehicle.m_flags));
-            MethodInfo logTraffic = typeof(IExtVehicleManager).GetMethod(nameof(IExtVehicleManager.LogTraffic));
+        public static void UpdatePosition(ushort vehicleId, ref Vehicle vehicleData) {
+            Constants.ManagerFactory.ExtVehicleManager.UpdateVehiclePosition(vehicleId, ref vehicleData);
+            if (SavedGameOptions.Instance.advancedAI && (vehicleData.m_flags & Vehicle.Flags.Spawned) != 0) {
+                // Advanced AI traffic measurement
+                Constants.ManagerFactory.ExtVehicleManager.LogTraffic(vehicleId, ref vehicleData);
+            }
+        }
 
+        internal static List<CodeInstruction> GetUpdatePositionInstructions() {
             return new List<CodeInstruction> {
-                new CodeInstruction(OpCodes.Call, getFactory.GetGetMethod()),
-                new CodeInstruction(OpCodes.Callvirt, getExtVehMan.GetGetMethod()),
-                new CodeInstruction(OpCodes.Ldarg_1),
-                new CodeInstruction(OpCodes.Ldarg_2),
-                new CodeInstruction(OpCodes.Callvirt, updateVehPos),
-                new CodeInstruction(OpCodes.Ldsfld, advAIField),
-                new CodeInstruction(OpCodes.Brfalse_S, label),
-                new CodeInstruction(OpCodes.Ldarg_2),
-                new CodeInstruction(OpCodes.Ldfld, vehFlags),
-                new CodeInstruction(OpCodes.Ldc_I4_4),
-                new CodeInstruction(OpCodes.And),
-                new CodeInstruction(OpCodes.Brfalse_S, label),
-                new CodeInstruction(OpCodes.Call, getFactory.GetGetMethod()),
-                new CodeInstruction(OpCodes.Callvirt, getExtVehMan.GetGetMethod()),
-                new CodeInstruction(OpCodes.Ldarg_1),
-                new CodeInstruction(OpCodes.Ldarg_2),
-                new CodeInstruction(OpCodes.Callvirt, logTraffic),
+                new CodeInstruction(OpCodes.Ldarg_1), // vehicleID
+                new CodeInstruction(OpCodes.Ldarga_S, 2), // ref vehicleData
```

Comment:
I've checked the original compiled code and there is no need for `load ref` instruction simply because argument is already `ref param` so there was no bug or mistake.

Reply:
Understood

Reply:
`Ldarg_2` is better then